internal/cloud_connector/message_handlers.go

```diff
@@ -29,9 +28,13 @@ const (
 	playbookWorkerDispatcherKey = "rhc-worker-playbook"
 )
 
-func HandleControlMessage(cfg *config.Config, mqttClient MQTT.Client, topicBuilder *mqtt.TopicBuilder, connectionRegistrar connection_repository.ConnectionRegistrar, accountResolver controller.AccountIdResolver, connectedClientRecorder controller.ConnectedClientRecorder, sourcesRecorder controller.SourcesRecorder) func(MQTT.Client, domain.ClientID, string) {
+// HandleControlMessage returns a function that processes control messages.
+// The returned function should only return an error in the case where the
+// message should get processed again.  In other words, if the message
```

Comment:
\<just curious\>if an error happens when handling a message - how are you planning on ensuring that the message gets processed again?

Reply:
The main reason I am looking to move toward explicitly committing the kafka messages is in the case where cloud-connector cannot reach the database.  I do not want to lose the connection-status messages in the case where the database is down for a bit.  I have taken a "simple" (maybe lazy is a better description) approach to handling that situation...cloud-connector simply falls over and waits for openshift to restart the pod.  In that case, cloud-connector will try to re-process the kafka message as it was not committed.

So with this approach to explicitly committing the messages and falling over in the case of not being able to contact the database, I hope to have lessened the chance of losing messages.

Regarding committing a higher offset on the same partition, I have built the kafka message consumer so that it is single threaded.  So with the "falling over approach" and the fact that the kafka message consumer is single threaded, is it still possible to commit a message with a higher offset?  To be honest, I'm still learning about kafka, its offsets and its partitions so I'll gladly take any help here.

Reply:
Ah ok. "Failing the pod" when e.g. message processing fails due to unreachable db is very reasonable IMO. I was not sure if that's you goal, as the code in `kafka_message_consumer.go` (specifically the continue keyword that moves on to process the next message) confused me (as processing of more messages could result in a higher offset getting committed, thus defeating the purpose of manual commit control). But now after your reply I assume that's just temporary code state given by the draft nature of this PR. Thanks for explanation.